### PR TITLE
CPDP-2944 net gain calculator

### PR DESCRIPTION
**Pull request by nubz**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposals/service/returns/CgtCalculationService.scala [hunk 1]
```diff
@@ -99,40 +100,23 @@ class CgtCalculationServiceImpl extends CgtCalculationService {
...
-      if (gainOrLossAfterLosses > AmountInPence.zero)
-        (gainOrLossAfterLosses -- exemptionAndLosses.annualExemptAmount).withFloorZero
-      else
-        gainOrLossAfterLosses
+    val TaxableGainOrLossCalculation(taxableGain, _, gainOrLossAfterInYearLosses, yearPosition, _, _) =
+      calculateTaxableGainOrLoss(
+        gainOrLossAfterReliefs,
+        exemptionAndLosses,
+        List.empty,
+        Address.UkAddress("stub", None, None, None, Postcode("ABC D12"))
```
[the remark this answers is not in the document]
Comment (nubz) [reply app/uk/gov/hmrc/cgtpropertydisposals/service/returns/CgtCalculationService.scala:108]: the calculator data wants current return address (for itemising gains on further return calcs) but for first returns we don't actually do anything with it, it gets thrown away - so not sure if we should go to the trouble of propagating the actual address through to here

[the remark this answers is not in the document]
Comment (nubz) [reply app/uk/gov/hmrc/cgtpropertydisposals/service/returns/CgtCalculationService.scala:108]: other alternative I guess is to make it an Option...

**Comment by rambabu-posa** [comment]
LGTM

**Comment by rambabu-posa** [comment]
LGTM

**Pull request merged**